### DISCUSSION TITLE
Add a button that ends/closes threads

### DISCUSSION THREAD

**Discussion by gigawhitlocks** [mattermost/community#21] [General Ideas]
Sometimes it can make sense to close a thread for further comments. It could be interesting if users (maybe only the thread starter?) in a channel could choose to deem that a thread is "finished" and end discussion in the thread, forcing participants to move back to the main channel or to start a new thread

An example scenario: 

1) a thread is started in a channel, and all of the channel participants start using the thread because the topic is of greater interest than any other conversation in the channel
2) after awhile, the conversation gets off topic
3) users would like to return to the main channel since the thread is no longer useful, so that they can reply (start new threads) to posts in the current conversation which is now off-topic in a thread containing everyone from the channel

Today:
4) someone points this out in the thread and everyone moves slowly back to the regular channel, but sometimes there are stragglers

Proposed:
4) someone clicks "close thread"
5) a message is posted to the end of the thread by the system notifying everyone that the thread has concluded, and the input box is disabled for everyone in the thread
6) everyone gets the message and moves back to the main channel (success!)

**Comment by esethna** [comment]
Interesting, I like this idea because it stops the problem of a thread topic evolving. 

Similar to this is the idea to pop-out a reply into the main channel so a new thread can be started off that post.

**Reply by gigawhitlocks** [reply]
>the idea to pop-out a reply into the main channel so a new thread can be started off that post.

oh I _love_ this idea

**Reply by Willyfrog** [reply]
that would be super useful! can this be done via wrangler?

**Reply by gigawhitlocks** [reply]
no, doesn't look like it. Wrangler assumes you want to move the whole thread

**Reply by Willyfrog** [reply]
> no, doesn't look like it. Wrangler assumes you want to move the whole thread

Sorry I probably wrote that too quickly.

For Eric's idea it might need some coding of the new feature, but it probably is doable

For closing a thread, maybe moving it to a read only/archived channel (as a PoC test and see how people feel about it)

**Comment by chetanyakan** [comment]
Another interesting idea could be thread splitting: allow the main thread to show children/related threads in UI somewhere.
So, people can follow related threads/topics together.
